pocs/tests/utils/google/test_storage.py

```diff
@@ -0,0 +1,111 @@
+import pytest
+import os
+import re
+import shutil
+
+from pocs.utils.config import load_config
+from pocs.utils.error import GoogleCloudError
+from pocs.utils.google.storage import PanStorage, upload_observation_to_bucket
+
+
+pytestmark = pytest.mark.skipif(
+    not pytest.config.option.test_cloud_storage,
+    reason="Needs --test-cloud-storage to run"
+)
+
+
+@pytest.fixture(scope="module")
+def auth_key():
+    local_config = load_config('pocs_local', ignore_local=True)
+    auth_key = os.path.join(
+        os.environ['PANDIR'],
+        '.keys',
+        local_config['panoptes_network']['auth_key']
+    )
+
+    return auth_key
+
+
+def test_key_exists(auth_key):
+    assert os.path.exists(auth_key), "No API key file"
+
+
+def test_bad_bucket(auth_key):
+    with pytest.raises(AssertionError):
+        PanStorage('fake-bucket')
+
+    with pytest.raises(GoogleCloudError):
+        PanStorage('fake-bucket', auth_key=auth_key)
+
+
+@pytest.fixture(scope="function")
+def storage(auth_key):
```

Comment:
I can't help but think that there should be a way that we can have an auth_key that is not in github for all the world to see, but that can be used by each tester for accessing the test bucket without having to use the same credentials that give access to the primary bucket. Of course, maybe you've already addressed this.

If not, travis-ci does support secret values being passed into tests.

Reply:
I think there is a generic testing auth_key. I saw reference to that somewhere, will try to find.